### PR TITLE
Fix firmware download command errors

### PR DESCRIPTION
**Pull request by xyan264**
Due to a limitation in Switchtec firmware, `fw-download` command can only support chunk size <1024 bytes. 

512 has been found to be the maximum chunk size that currently works with `fw-download` command.

### CODE REVIEW

File: nvme.c [hunk 1]
```diff
@@ -2318,8 +2318,8 @@ static int fw_download(int argc, char **argv, struct command *cmd, struct plugin
 	}
 
 	buf = fw_buf;
-	if (cfg.xfer == 0 || cfg.xfer % 4096)
-		cfg.xfer = 4096;
+	if (cfg.xfer == 0 || cfg.xfer % 512 || cfg.xfer > 512)
```
Comment (kelvin-cao) [inline nvme.c:2320]: Does it mean for whatever cfg.xfer's value, we'll assign it to 512?

Comment (xyan264) [reply nvme.c:2320]: Yes that's what it does. I wanted to change it to `cfg.xfer=512` but decided to keep it in case we support other size later.

**Review by kelvin-cao — APPROVED** [review]
thanks.

File: nvme.c (hunk 1 above)
Comment (xyan264) [reply nvme.c]: Updated with `cfg.xfer=512`.

**Pull request merged**